frontend/src/metabase-types/api/card.ts

```diff
@@ -92,45 +110,188 @@ export type TableColumnOrderSetting = {
   field_ref?: FieldReference;
 };
 
-export type VisualizationSettings = {
-  "graph.show_values"?: boolean;
-  "stackable.stack_type"?: "stacked" | "normalized" | null;
-
-  // Table
-  "table.columns"?: TableColumnOrderSetting[];
+export type ButtonVariant =
+  | "primary"
+  | "default"
+  | "danger"
+  | "success"
+  | "borderless";
 
-  // X-axis
-  "graph.x_axis.title_text"?: string;
-  "graph.x_axis.scale"?: "ordinal";
-  "graph.x_axis.axis_enabled"?: "compact";
-
-  // Y-axis
-  "graph.y_axis.title_text"?: string;
-  "graph.y_axis.scale"?: "linear" | "pow" | "log";
-  "graph.y_axis.axis_enabled"?: true;
+export type BasicVisualizationSettings = {
```

Comment:
I'm not sure if it is valuable to extract "Basic" visualization settings here. They look no different from not basic like `axis` or `show_mini_bar` — these are visualization-speficic settings as `funnel.type`. Could you please describe the logic behind the split?

I believe to make a split we could do something like:
```
type PieVisualizationSettings = {
  // pie specific settings
}

type FunnelVisualizationSettings = {
  // funnel specific settings
}

type CommonVisualizationSettings = {
  display: ...
}

type DashcardVisualizationSettings = {
  click_behavior: {...}
}

type VisualizationSettings = PieVisualizationSettings | FunnelVisualizationSettings | CommonVisualizationSettings | DashcardVisualizationSettings
```

What do you think?